parquet-variant-compute/src/variant_get.rs

```diff
@@ -142,8 +142,17 @@ fn shredded_get_path(
             for i in 0..target.len() {
                 if target.is_null(i) {
                     builder.append_null()?;
+                } else if !cast_options.safe {
+                    let value = target.try_value(i)?;
+                    builder.append_value(value)?;
                 } else {
-                    builder.append_value(target.value(i))?;
+                    let _ = match target.try_value(i) {
+                        Ok(v) => builder.append_value(v)?,
+                        Err(_) => {
+                            builder.append_null()?;
+                            false // add this to make match arms have the same return type
+                        }
+                    };
```

Comment:
This might be a bit better as it drops the value early
```suggestion
                    match target.try_value(i) {
                        Ok(v) => {
                            let _ = builder.append_value(v)?;
                        }
                        Err(_) => builder.append_null()?,
                    }
```

Reply:
I tried this, but changed to the current solution.
We need to change the return type for `builder.append_null()`, (we need both match arms to return the same type), and we need to return `Ok(false)` for `builder.append_null()` to satisfy the semantics here, but the `builder.append_null` always succeeds(`builder.append_value()` will return `Ok(true)` if succeed). It seems weird to have different semantics for the return result of `builder.append_value` and `builder.append_null`.